osc/osc_test.go

```diff
@@ -111,53 +113,80 @@ func TestAddMsgHandlerWithInvalidAddress(t *testing.T) {
 	}
 }
 
+// These tests stop the server by forcibly closing the connection, which causes
+// a "use of closed network connection" error the next time we try to read from
+// the connection. As a workaround, this wraps server.ListenAndServe() in an
+// error-handling layer that doesn't consider "use of closed network connection"
+// an error.
+//
+// Open question: is this desired behavior, or should server.serve return
+// successfully in cases where it would otherwise throw this error?
+func serveUntilInterrupted(server *Server) error {
+	if err := server.ListenAndServe(); err != nil &&
+		!strings.Contains(err.Error(), "use of closed network connection") {
+		return err
+	}
+
+	return nil
+}
+
 func TestServerMessageDispatching(t *testing.T) {
 	finish := make(chan bool)
 	start := make(chan bool)
 	done := sync.WaitGroup{}
 	done.Add(2)
 
-	// Start the OSC server in a new go-routine
-	go func() {
-		conn, err := net.ListenPacket("udp", "localhost:6677")
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer conn.Close()
+	port := 6677
+	addr := "localhost:" + strconv.Itoa(port)
+
+	server := &Server{Addr: addr, Dispatcher: NewStandardDispatcher()}
+	defer server.CloseConnection()
```

Comment:
There's no need for this call with the way you are closing the connection inside the handler.
We would end up doing it twice and not checking the response value.

Reply:
Good point! I'll remove this line and handle the error in the place where I'm calling `server.CloseConnection()` further down.